pcsx2/GS/Renderers/HW/GSRendererNew.cpp

```diff
@@ -1470,6 +1486,10 @@ void GSRendererNew::DrawPrims(GSTexture* rt, GSTexture* ds, GSTextureCache::Sour
 		m_conf.require_full_barrier = false;
 	}
 
+	// Don't emit the second color output from the pixel shader when it's
+	// not going to be used (no or sw blending).
+	m_conf.ps.no_color1 |= (m_conf.blend.index == 0);
```

Comment:
Mind making this fully check the blend mode and set this flag for all cases where dual source blend isn't used?  Intel GPUs on Metal (and MVK) have a driver bug where they don't write depth if dual source blend outputs go unused.